.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,56 @@
+name: build_commits
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: Check if build works for every commit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure git
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "github-actions@users.noreply.github.com"
```

Comment:
```
Run git config user.name "GitHub Actions"
  git config user.name "GitHub Actions"
  git config user.email "github-actions@users.noreply.github.com"
  shell: /usr/bin/bash -e {0}
fatal: not in a git directory
Error: Process completed with exit code 128.
```
Ah, my bad, I should have foreseen this.

You can either add `--global` to the commands or move the configuration after the checkout, either should work.

Reply:
I tried both running with the --global in my other repo and after the checkout but was still getting the same error.

Reply:
I tried both running with the --global in my other repo and after the checkout but was still getting the same error.

Reply:
It worked! Mistake: Previously I was configuring git after checkout and was running it with the global option.